### PR TITLE
Add cleanup script for npm publish

### PR DESCRIPTION
**Pull request by 0xaptosj**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -18,6 +18,7 @@
   "scripts": {
     "clean": "rm -rf dist",
     "build": "npm run clean && tsc && cp -r templates dist/templates && cp -r examples dist/examples",
+    "publish-to-npm": "./cleanup_for_publish.sh && pnpm i && pnpm build && npm publish",
```
Comment (0xmaayan) [inline package.json:21]: we use `npm` as the package manage, all of our scripts use `npm`
```suggestion
    "publish-to-npm": "./cleanup_for_publish.sh && npm install && npm run build && npm publish",
```

Comment (0xaptosj) [reply package.json:21]: hmm i thought for CAD itself, we use pnpm?

Comment (0xaptosj) [reply package.json:21]: then ill remove the pnpm lock?

File: README.md [hunk 2]
```diff
@@ -46,3 +46,11 @@ Then follow the prompts!
 `create-aptos-dapp` also provides you with full stack examples that are for educational purpose, these examples are not production ready and not audited.
 
 - **Aptos Friend Template:** A friend tech style social app that implements share trading
+
+### Releasing a new version
+
+Run this to publish a new version to npm:
+
+```bash
+npm run publish-to-npm
```
Comment (0xmaayan) [inline README.md:55]: a crucial step is to run `npm publish --dry-run` before running `npm publish`

Lets:
```
npm run prepare-to-publish
npm publish --dry-run
// make sure release info is aligned with what is currently published on npm
npm publish
```

File: package.json [hunk 3]
```diff
@@ -18,6 +18,7 @@
   "scripts": {
     "clean": "rm -rf dist",
     "build": "npm run clean && tsc && cp -r templates dist/templates && cp -r examples dist/examples",
+    "publish-to-npm": "./cleanup_for_publish.sh && npm install && npm run build && npm publish",
```
Comment (0xmaayan) [inline package.json:21]: a crucial step is to run `npm publish --dry-run` before running npm publish


```suggestion
    "publish-to-npm": "./cleanup_for_publish.sh && npm install && npm run build",
```

Comment (0xmaayan) [inline package.json:21]: what do you need `npm install` for? you need `npm run build`

Comment (0xaptosj) [reply package.json:21]: when i run `npm run build` without node_modules, i got error

File: README.md [hunk 4]
```diff
@@ -46,3 +46,23 @@ Then follow the prompts!
...
+
+### Releasing a new version
+
+Run this to cleanup the project, build it and prepare it for publishing:
+
+```bash
+npm run prepare-to-publish
+```
+
+Run this to preview the change:
```
Comment (0xmaayan) [inline README.md:58]:
```suggestion
Run this to preview the change and make sure it aligns with [the existing published package](https://www.npmjs.com/package/create-aptos-dapp):
```

**Review by 0xmaayan — APPROVED** [review]

**Pull request merged**